copy/single.go

```diff
@@ -173,7 +189,7 @@ func (c *copier) copySingleImage(ctx context.Context, unparsedImage *image.Unpar
 	ic.diffIDsAreNeeded = src.UpdatedImageNeedsLayerDiffIDs(*ic.manifestUpdates)
 
 	// If enabled, fetch and compare the destination's manifest. And as an optimization skip updating the destination iff equal
-	if c.options.OptimizeDestinationImageAlreadyExists {
+	if c.options.OptimizeDestinationImageAlreadyExists && !ic.requireCompressionFormatMatch{
```

Comment:
Stylistically I’d prefer this to be one of the checks inside this top-level `if`, so that the “checking if we can skip copying” line is logged (and that should log the new value as well).